src/main/java/com/onevizion/scmdb/PackageGenerator.java

```diff
@@ -0,0 +1,502 @@
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.onevizion.scmdb.ColorLogger.Color.*;
+
+@Component
+public class PackageGenerator {
+    private static final String PACKAGE_SPECIFICATION_SUFFIX = "_spec";
+    private static final String ROLLBACK_SUFFIX = "_rollback";
+    private static final String PACKAGES_DDL_DIRECTORY_NAME = "packages";
+
+    private Git git;
+    private Repository repository;
+
+    @Autowired
+    private DbScriptFacade scriptFacade;
+
+    @Autowired
+    private AppArguments appArguments;
+
+    @Autowired
+    private ColorLogger logger;
+
+    public void generatePackages() {
+        try {
+            git = Git.open(getRepositoryDirectory());
+            repository = git.getRepository();
+
+            List<String> changedFilesList = getChangedPackages();
+            if (CollectionUtils.isEmpty(changedFilesList)) {
+                logger.info("Not found changes in packages");
+            } else {
+                generateScripts(changedFilesList);
+                return;
+            }
+
+            RevCommit mergeCommit = getMergeCommit();
+            if (mergeCommit == null) {
+                logger.info("Merge commit `master` into {} not found", repository.getBranch());
+            } else {
+                updateScript(mergeCommit);
+            }
+
+        } catch (GitAPIException | IOException e) {
+            logger.error("Error with Git: ", e);
+        } finally {
+            git.close();
+        }
+    }
+
+    public void generateScripts(List<String> changedFilesList) {
+        try {
+            logger.info("Generating scripts for packages: " + changedFilesList);
+            boolean wasSpec = false;
+            String packageName = "";
+            String scriptCommitName;
+            String scriptRollbackName;
+
+            createStash();
+
+            for (String fileName : changedFilesList) {
+                if (fileName.contains("_spec.sql")) {
+                    packageName = getPackageNameFromFile(fileName, true);
+                    scriptCommitName = createCommitScript(packageName);
+                    scriptRollbackName = createRollbackScript(packageName);
+                    wasSpec = true;
+                } else if (!wasSpec || !fileName.contains(packageName)) {
+                    packageName = getPackageNameFromFile(fileName, false);
+                    scriptCommitName = createCommitScript(packageName);
+                    scriptRollbackName = createRollbackScript(packageName);
+                    wasSpec = false;
+                } else {
+                    wasSpec = false;
+                    continue;
+                }
+                git.add().addFilepattern("db/scripts/" + scriptCommitName).call();
+                git.add().addFilepattern("db/scripts/" + scriptRollbackName).call();
+            }
+            deleteStash();
+            git.close();
```

Comment:
you already have `git.close` call in the `finally` section of the `generatePackages`